docs/source/tutorial/basic.rst

```diff
@@ -2,21 +2,586 @@
+
+                builder = self.inputs.code.get_builder()
+
+                builder.x = self.ctx.multiple
+                builder.y = self.inputs.z
+
+                future = self.submit(builder)
+
+                return ToContext({'addition': future})
+
+            def result(self):
+                self.out('result', self.ctx['addition'].get_outgoing().get_node_by_label('sum'))
+
+    First, we recognize the ``multiply`` function we have used earlier, decorated as a ``calcfunction``.
+    The ``define`` class method specifies the ``input`` and ``output`` of the ``WorkChain``, as well as the ``outline``, which are the steps of the workflow.
+    These steps are provided as methods of the ``MultiplyAddWorkChain`` class.
+
+.. note::
+
+    Besides WorkChain's, workflows can also be implemented as *work functions*.
+    These are ideal for workflows that are not very computationally intensive and can be easily implemented in a Python function.
+
+Let's run the ``WorkChain`` above!
+Start up the ``verdi shell`` and import the ``MultiplyAddWorkChain``:
+
+.. code-block:: ipython
+
+    In [1]: from aiida.workflows.multiplyadd import MultiplyAddWorkChain
+
+Similar to a ``CalcJob``, the ``WorkChain`` input can be set up using a builder:
+
+.. code-block:: ipython
+
+    In [2]: builder = MultiplyAddWorkChain.get_builder()
+       ...: builder.code = load_code(label='add')
+       ...: builder.x = Int(2)
+       ...: builder.y = Int(3)
+       ...: builder.z = Int(5)
+
+Once the ``WorkChain`` input has been set up, we submit it to the daemon using the ``submit`` function from the AiiDA engine:
+
+.. code-block:: ipython
+
+    In [3]: from aiida.engine import submit
+       ...: submit(builder)
+
+Now quickly leave the IPython shell and check the process list:
+
+.. code-block:: console
+
+    $ verdi process list -a
+
+Depending on which step the workflow is running, you should get something like the following:
+
+.. code-block:: bash
+
+      PK  Created    Process label             Process State    Process status
+    ----  ---------  ------------------------  ---------------  ------------------------------------
+       3  7m ago     multiply                  ⏹ Finished [0]
+       7  3m ago     ArithmeticAddCalculation  ⏹ Finished [0]
+      12  2m ago     ArithmeticAddCalculation  ⏹ Finished [0]
+      19  16s ago    MultiplyAddWorkChain      ⏵ Waiting        Waiting for child processes: 22
+      20  16s ago    multiply                  ⏹ Finished [0]
+      22  15s ago    ArithmeticAddCalculation  ⏵ Waiting        Waiting for transport task: retrieve
+
+    Total results: 6
+
+    Info: last time an entry changed state: 0s ago (at 09:08:59 on 2020-05-13)
+
+We can see that the ``MultiplyAddWorkChain`` is currently waiting for its *child process*, the ``ArithmeticAddCalculation``, to finish.
+Check the process list again for *all* processes (You should know how by now!).
+Now all the processes should be in the ``Finished`` state.
```

Comment:
Meaning we should wait until they all finish or should they be all finished after checking the processes?